Sources/Shared/TCPSocket.cpp

```diff
@@ -42,7 +42,9 @@ int TCPSocket::send(const std::string &data) {
   return ::send(this->socketDescriptor, data.c_str(), data.size(), 0);
 }
 
-int TCPSocket::receive(std::string &data) {
+int TCPSocket::receive(std::string &data, int bytes) {
+  char buffer[bytes];
```

Comment:
1. I know.
2. GCC and Clang support Variable-Length Arrays via an extension. However, VLAs like char array[size] are not part of the C++11 (or C++14 for that matter) standard and their use is heavily discouraged for a lot of reasons (for example because such an array deprecates to a simple pointer whenever passed to a function and doesn't know its size).